lib/driver-factory.js

```diff
@@ -1,23 +1,26 @@
+const parseVersion = require('./util/parse-version');
+const Driver = require('./driver');
+const cqlTypes = require('./util/cql-types');
 
-
-var parseVersion = require('./util/parse-version');
-var Driver = require('./driver');
-var cqlTypes = require('./util/cql-types');
-
-function DriverFactory(context) {
+function driverFactory(context) {
   context = context || {};
   context.config = context.config || {};
-  context.config.version = context.config.cqlVersion = context.config.cqlVersion || context.config.version || '3.1.0';
-  context.config.parsedCqlVersion = parseVersion(context.config.cqlVersion);
+  const {
+    protocolOptions: {
+      maxVersion = '3.1.0'
+    } = {}
+  } = context.config;
+  context.config.version = maxVersion;
+  context.config.parsedCqlVersion = parseVersion(maxVersion);
 
-  var protocol = context.config.protocol || 'binary';
+  const protocol = context.config.protocol || 'binary';
```

Comment:
`binary` here is also unnecessary. The current driver is binary-only. `helenus` was plain-text, which is where this comes from.

It's possible this `protocol` variable can be entirely dropped.

Reply:
@scommisso: yep, looks to be unused. Removed.